lib/lotus/model/adapters/sql/command.rb

```diff
@@ -55,10 +59,41 @@ def update(entity)
           # @api private
           # @since 0.1.0
           def delete
-            @collection.delete
+            _rescue_database_error do
+              @collection.delete
+            end
           end
 
           alias_method :clear, :delete
+
+          private
+
+          # Rescues Sequel::DatabaseError in yielded block
+          #
+          # @api private
+          def _rescue_database_error
```

Comment:
in theory private function should not be tested. But this function is an exception due to its richness in logic. I think you should remove the `private` syntax.